content/2015.08.donnees-et-signatures-en.rst

```diff
@@ -0,0 +1,144 @@
+Sign collection to enforce data integrity
+#########################################
+
+:date: 2015-08-26
+:summary: How to enforce data integrity to Kinto users using collection hash signature.
+:slug: data-signature
+:lang: en
+:url: data-signature
+
+.. note::
+
+    Translated from the French by `Rémy Hubscher <https://twitter.com/Natim>`_.
+
+Within the scope of the `Go Faster <https://wiki.mozilla.org/Firefox/Go_Faster>`_
+project, we want to distribute minor Firefox updates more often than
+just with major updates (which happens once every six weeks).
+
+There are multiple kind of data that we will need to update on the client.
+Among others, we would like to `update the SSL OneCRL (Control Revocation List)
+<https://blog.mozilla.org/security/2015/03/03/revoking-intermediate-certificates-introducing-onecrl/>`_.
+
+In that specific case, it is obviously very important to make sure
+that all the fetched data are coming from Mozilla and that no records
+are missing. So that we can be absolutely confident that nobody added
+a valide certificate to the CRL or remove a revocated one.
+
+A cryptographic signature can give us the guarantee that all the
+records where fetched but it is still possible to prevent users to
+completely access the service (*Chinese Great Firewall* for instance.)
+
+This mecanism works in our specific use-cases of the OneCRL update
+project, but it can also be useful to any future project that needs to
+make sure that all records on a collection have been correctly synced.
+We will probably need to re-use it to update other part of Firefox or
+Fennec but you may also want to use it for your use-cases.
+
+In order to distribute the data or meta-data associated with the files
+that needs to be updated, we plan to use `Kinto
+<https://kinto.readthedocs.org>`_. Kinto is a good fit because it can
+easily be cached behind a CDN.
```

Comment:
This isn't part of the original article. I'm not actually sure the fact it's easily cachable makes kinto a good fit here. The fact that it handles synchronisation out the box probably would be a better thing to point out.